src/components/game/UpdateGameForm.js

```diff
@@ -0,0 +1,118 @@
+import { useState, useEffect } from "react"
+import { useNavigate, useParams } from 'react-router-dom'
+import { getGameTypes, getSingleGame, updateGame } from '../../managers/GameManager.js'
+
+
+
+export const UpdateGameForm = () => {
+
+    const { gameId } = useParams()
+
+    const [currentGame, setCurrentGame] = useState({})
+
+    const [gameTypes, setGameTypes] = useState([])
+
+    const navigate = useNavigate()
+
+
+    useEffect(
+        () => {
+            getSingleGame(gameId).then(game => setCurrentGame(game))
+            getGameTypes().then(type => setGameTypes(type))
+        }, [gameId])
+
+    const changeGameState = (domEvent) => {
+        const newGame = Object.assign({}, currentGame)
```

Comment:
You said you didn't understand what this is doing. It's exactly the same as `{ ...currentGame }`, which you are using elsewhere. It creates an empty object, then assigns each prop of `currentGame` to the new object. It creates a shallow clone of `currentGame`.